Spec/Tests/PushActivationStateMachineTests.swift

```diff
@@ -1264,13 +1273,10 @@ class PushActivationStateMachineTests: XCTestCase {
 
             expect(stateMachine.current).to(beAKindOf(ARTPushActivationStateNotActivated.self))
             expect(httpExecutor.requests.count) == 1
-            let requests = httpExecutor.requests.compactMap { $0.url?.path }.filter { $0 == "/push/deviceRegistrations/\(rest.device.id)" }
+            let requests = httpExecutor.requests.filter({ $0.url?.path == "/push/deviceRegistrations/\(deviceId)" })
             expect(requests).to(haveCount(1))
-            guard let request = httpExecutor.requests.first else {
-                fail("should have a \"/push/deviceRegistrations\" request"); return
-            }
-            guard let url = request.url else {
-                fail("should have a \"/push/deviceRegistrations\" URL"); return
+            guard let request = requests.first, let url = request.url else {
+                fail("should have a '/push/deviceRegistrations/\(deviceId)' request"); return
```

Comment:
This message is not exactly true, because `url` could be `nil`.
Maybe this is a good moment to change this guard to `XCTUnwrap`?

Reply:
I guess it was changed in the PR for `XCTUnwrap`.